pinot-core/src/main/java/org/apache/pinot/core/transport/server/routing/stats/ServerRoutingStatsManager.java

```diff
@@ -0,0 +1,374 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.transport.server.routing.stats;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ *
+ *  {@code ServerRoutingStatsManager} manages the query routing stats for each server. The stats are maintained at
+ *  the broker and are updated when a query is submitted to a server and when a server responds after processing a
+ *  query.
+ */
+public class ServerRoutingStatsManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServerRoutingStatsManager.class);
+
+  private final PinotConfiguration _config;
+  private volatile boolean _isEnabled;
+  private ConcurrentHashMap<String, ServerRoutingStatsEntry> _serverQueryStatsMap;
+  private ExecutorService _executorService;
+
+  private double _alpha;
+  private long _autoDecayWindowMs;
+  private long _warmupDurationMs;
+  private double _avgInitializationVal;
+  private int _c3ScoreExponent;
+
+  public ServerRoutingStatsManager(PinotConfiguration pinotConfig) {
+    _config = pinotConfig;
+  }
+
+  public void init() {
+    _isEnabled = _config.getProperty(CommonConstants.Broker.AdaptiveServerSelector.CONFIG_OF_ENABLE_STATS_COLLECTION,
+        CommonConstants.Broker.AdaptiveServerSelector.DEFAULT_ENABLE_STATS_COLLECTION);
```

Comment:
Suggest importing `CommonConstants.Broker.AdaptiveServerSelector` to make code a bit more readable

Reply:
Done